sklearn/utils/multiclass.py

```diff
@@ -266,9 +269,14 @@ def type_of_target(y):
         pass
 
     # Invalid inputs
-    if y.ndim > 2 or (y.dtype == object and len(y) and
```

Comment:
I am confused. I don't understand why `len` should be used when the dtype is int64 etc... doesn't the error imply that `y.dtype == object`?

Reply:
I kept it because I couldn't figure it out, but actually you are right, all provided examples in tests work fine without the `len(y)` condition

Reply:
FYI I remove the len() condition

Reply:
I guess the `len` condition must have been used for empty or 0-dim ndarrays. Anyway I think I covered all possible cases.